unison-share-api/src/Unison/Server/CodebaseServer.hs

```diff
@@ -475,21 +457,23 @@ startServer env opts rt codebase onStart = do
   envUI <- canonicalizePath $ fromMaybe (FilePath.takeDirectory exePath </> "ui") (codebaseUIPath opts)
   token <- case token opts of
     Just t -> return $ C8.pack t
-    _ -> genToken
+    Nothing -> return $ C8.pack "codebase"
   let baseUrl = BaseUrl (fromMaybe "http://127.0.0.1" (host opts)) token
   let settings =
         defaultSettings
-          & maybe id setPort (port opts)
-          & maybe id (setHost . fromString) (host opts)
-  let a = app env rt codebase envUI token (allowCorsHost opts)
+          & maybe (setPort 5858) setPort (port opts)
+          & maybe (setHost $ fromString "127.0.0.1") (setHost . fromString) (host opts)
```

Comment:
This looks good.  If you want, I think you can also factor out the `setPort`/`setHost.fromString` like:

```suggestion
          & setPort (fromMaybe 5858 (port opts))
          & (setHost . fromString) (fromMaybe "127.0.0.1" (host opts))
```
(written up by hand, not typechecked)

Reply:
There's still the separate issue that I worry might break something if `UCM_HOST` is actually set. @ChrisPenner 

But it's not related to this PR, so I'm ready to merge once you decide on the above suggestion. @hojberg

Reply:
Currently it looks like it needs UCM_HOST to be an ip to work (it will crash with a `Network.Socket.bind: unsupported operation (Can't assign requested address)` if given a url).

I think this should be dealt with separately.